custom/enikshay/tests/test_adherence_updates.py

```diff
@@ -0,0 +1,363 @@
+                self.episode_id,
+                adherence_date,
+                extra_update={
+                    "name": adherence_date,
+                    "adherence_value": adherence_value,
+                }
+            )
+            for (adherence_date, adherence_value) in adherence_cases
+        ])
+
+    def assert_update(self, input, output):
+        update = self.calculate_adherence_update(input)
+        self.assertDictEqual(
+            update.update_json(),
+            output
+        )
+
+    def calculate_adherence_update(self, input):
+        self.case_updater.purge_date = pytz.UTC.localize(input[0])
+        # setup episode and adherence cases
+        adherence_schedule_date_start, adherence_schedule_id = input[1]
+        adherence_cases = input[2]
+        episode = self._create_episode_case(adherence_schedule_date_start, adherence_schedule_id)
+        self._create_adherence_cases(adherence_cases)
+
+        return EpisodeUpdate(episode, self.case_updater)
+
+    def test_adherence_schedule_date_start_late(self):
+        #   Sample test case
+        #   [
+        #       (
+        #           purge_date,
+        #           (adherence_schedule_date_start, adherence_schedule_id),
+        #           [
+        #               (adherence_date, adherence_value),
+        #               (adherence_date, adherence_value),
+        #               ...
+        #           ],
+        #           {
+        #               'aggregated_score_date_calculated': value
+        #               'expected_doses_taken': value
+        #               'aggregated_score_count_taken': value
+        #           }
+        #       ),
+        #       ...
+        #   ]
+
+        self.assert_update(
+            (
+                datetime(2016, 1, 15),
+                (datetime(2016, 1, 17), 'schedule1'),
+                []
+            ),
+            {
+                'aggregated_score_date_calculated': date(2016, 1, 16),
+                'expected_doses_taken': 0,
+                'aggregated_score_count_taken': 0,
+                # 1 day before should be adherence_schedule_date_start,
+                'adherence_latest_date_recorded': date(2016, 1, 16),
+                'adherence_total_doses_taken': 0
+            }
+        )
+
+    def test_no_adherence_schedule_date_start(self):
+        self.assert_update(
+            (
+                datetime(2016, 1, 17),
+                (None, 'schedule1'),
+                []
+            ),
+            {
+            }
+        )
+
+    def test_no_adherence_cases(self):
+        self.assert_update(
+            (
+                datetime(2016, 1, 20),
+                (datetime(2016, 1, 10), 'schedule1'),
+                []
+            ),
+            {   # change
+                'aggregated_score_date_calculated': date(2016, 1, 20),
+                'expected_doses_taken': (10.0 / 7) * int(self.fixture_data['schedule1']),
```

Comment:
I think this should be 0 here. Since there's no adherence data there's no expected doses. 

The "10" in this case should ~always match (adherence_latest_date_recorded - adherence_schedule_date_start)